### ISSUE TITLE
Uncaught RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by nejlyn** [Floobits/floobits-atom#75]
I just installed floobits today. I first saw the error after the following steps:
1. Install floobits plugin
2. Create New Workspace with View & Edit Permissions
3. After ~30 seconds, the RangeError appears

After this, I tried closing/re-opening Atom and creating a workspace again, although this time I unchecked "Create New Workspace" since the workspace from my initial attempt had been created already. The error also appeared after doing this.

My colleague, who has a similar dev environment, is also experiencing the same issue, though on v0.21.1.

Thanks!

**Atom Version**: 1.0.19
**System**: Mac OS X 10.10.5
**Thrown From**: [floobits](https://github.com/Floobits/floobits-atom) package, v0.21.2
### Stack Trace

Uncaught RangeError: Maximum call stack size exceeded

```
At path.js:19

RangeError: Maximum call stack size exceeded
    at Array.join (native)
    at Object.posix.resolve (path.js:429:52)
    at Object.posix.relative (path.js:488:14)
    at to_rel_path (/Users/jennyhiggins/.atom/packages/floobits/lib/utils.js:72:28)
    at Object.is_shared (/Users/jennyhiggins/.atom/packages/floobits/lib/utils.js:79:10)
    at Collection.flux.createCollection.findFluffer (/Users/jennyhiggins/.atom/packages/floobits/lib/common/buffer_model.js:194:15)
    at /Users/jennyhiggins/.atom/packages/floobits/lib/common/handlers/floohandler.js:327:33
    at /Users/jennyhiggins/.atom/packages/floobits/node_modules/async/lib/async.js:187:20
    at replenish (/Users/jennyhiggins/.atom/packages/floobits/node_modules/async/lib/async.js:322:21)
    at /Users/jennyhiggins/.atom/packages/floobits/node_modules/async/lib/async.js:329:29
```
### Commands

```
     -0:32.7.0 command-palette:toggle (div.settings-view.pane-item)
  5x -0:30.8.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:29.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:29.2.0 Floobits: Create Workspace (div.settings-view.pane-item)
     -0:13 tree-view:show (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "whitespace"
    ],
    "packagesWithKeymapsDisabled": [
      "close-other-tabs"
    ],
    "ignoredNames": [
      "*-tests/*",
      "*-test/*",
      "*-Test*"
    ]
  }
}
```
### Installed Packages

``` coffee
# User
Sublime-Style-Column-Selection, v1.3.0
auto-reveal-active-file, v0.3.0
build, v0.47.0
close-other-tabs, v0.1.0
floobits, v0.21.2

# Dev
No dev packages
```

**Comment by ggreer** [comment]
Thanks for reporting this bug. This is very confusing. We're only passing two arguments to `path.relative()`, and they should be nowhere close to the limits of Chromium's array functions.

Can you tell me more about the path you're trying to share? Do you have a symlink loop or something?

**Comment by carcinocron** [comment]
Being totally unqualified to make this guess, but based on my encounter with this bug, your coworker may have shared something that either had too many files in one directory or just had too many files for floobits. (Maybe flowbits needs a sanely sized "Project Folder"?). I've barely started debugging this BTW.

**Comment by carcinocron** [comment]
My coworker shared a smaller sub directory and it seems to have worked out better. If the issue is too many files, too many deeply nested directories, or too many files in one directory, you might need to look into putting in a warning for that or switching from using the native array to something like collectionjs, generators, or something that relies on less recursion.

**Comment by kans** [comment]
@InstanceOfMichael 

If you look at the call stack, it is a nodejs std lib call [path.resolve](https://nodejs.org/api/path.html#path_path_resolve_from_to) that explodes.  You are the first person to have this problem - is it possible that you have a circular sym link?

**Comment by carcinocron** [comment]
When my coworker choose a more relevant sub directory it worked fine (his Project Folder was probably his whole home folder at the time).

I might have choosen /tmp/ as my local directory (because at the time, I was not aware that I was suppose to already have my own copy of the code we were collaborating on). Sorry for not reporting that detail earlier.

> is it possible that you have a circular sym link?

This is possible, but likely a conditional result of the entire /home/ or using the /tmp/. Definitely would not be there intentionally, as we do not use that explicitly for our project.

**Comment by ggreer** [comment]
The latest version (v0.24.1) should have this fixed. It stops recursing once it gets to depth 25. Symlink loops will still cause multiple copies of the same file to be uploaded, but at least the plugin won't crash.

**Comment by carcinocron** [comment]
It looks like its happening again.

This command says I don't have any symbolic links in the project directory.

```
sudo ls -la / | grep "\->"
```

**Atom Version**: 1.0.19
**System**: Ubuntu 14.04.3
**Thrown From**: [floobits](https://github.com/Floobits/floobits-atom) package, v0.25.0

### Stack Trace

Uncaught RangeError: Maximum call stack size exceeded

```
At path.js:19

RangeError: Maximum call stack size exceeded
    at Array.join (native)
    at Object.posix.resolve (path.js:429:52)
    at Object.posix.relative (path.js:488:14)
    at to_rel_path (/home/forge/.atom/packages/floobits/lib/utils.js:72:28)
    at Object.is_shared (/home/forge/.atom/packages/floobits/lib/utils.js:79:10)
    at Collection.flux.createCollection.findFluffer (/home/forge/.atom/packages/floobits/lib/common/buffer_model.js:194:15)
    at /home/forge/.atom/packages/floobits/lib/common/handlers/floohandler.js:335:33
    at /home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:187:20
    at replenish (/home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:322:21)
    at /home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:329:29
    at /home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:44:16
    at /home/forge/.atom/packages/floobits/lib/common/handlers/floohandler.js:345:18
    at /home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:187:20
    at replenish (/home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:322:21)
    at /home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:329:29
    at /home/forge/.atom/packages/floobits/node_modules/async/lib/async.js:44:16
```

### Commands

```
     -2:38.9.0 Floobits: Join Workspace (atom-text-editor.editor.is-focused)
     -0:42 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -0:33.1.0 command-palette:toggle (atom-text-editor.editor.is-focused)
  4x -0:29.7.0 core:move-down (atom-text-editor.editor.mini.is-focused)
  2x -0:28.8.0 core:move-up (atom-text-editor.editor.mini.is-focused)
     -0:28 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:28 Floobits: Join Workspace (atom-text-editor.editor)
     -0:06.6.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

``` json
{
  "core": {
    "disabledPackages": [
      "font-awesome-snippetset",
      "linter-phpmd"
    ]
  }
}
```

### Installed Packages

``` coffee
# User
aligner, v0.17.0
aligner-php, v1.0.0
bootstrap3-snippets, v0.1.2
drunken-php, v0.2.1
floobits, v0.25.0
jshint, v1.8.1
linter, v1.8.1
linter-php, v1.0.5
php-cs-fixer, v2.5.1

# Dev
No dev packages
```

**Comment by carcinocron** [comment]
Does floobits read my `.gitignore` files?

I added some logical entries to my `.flooignore` and corrected some invalid entries on my `.gitignore`s and it seems to be working now. An example of an invalid `.gitignore` entry:

```
/public/assets
```

which should actually have been:

```
public/assets
```

Edit: It's not working for my main project.

**Comment by kans** [comment]
@InstanceOfMichael  yes, it does

We just pushed the fix in v0.28.1.  Would you give it another try and let us know if it works for you?
